packages/waku/src/router/client.ts

```diff
@@ -216,6 +216,30 @@ export function useRouter() {
   };
 }
 
+function useSharedRef<T>(
+  ref:
+    | React.RefObject<T | null>
+    | ((instance: T | null) => void)
+    | null
+    | undefined,
+): React.RefObject<T | null> {
+  const managedRef = useRef<T>(null);
+
+  return {
+    set current(value: T | null) {
+      managedRef.current = value;
+      if (typeof ref === 'function') {
+        ref(value);
```

Comment:
Ref callback may return a cleanup function. We should handle it too.

Reply:
This change should handle cleanup functions being returned by callback refs: https://github.com/wakujs/waku/pull/1455/commits/16c9015f07f828d1d8856ebb8f94d8451ceb911d

Unfortunately I couldn't find a way to preserve the return type of `useSharedRef` since `setters` can't return a value - so I changed it to look a bit like `useState`, returning the managed ref and a function to set the ref.